lib/pbench/server/database/models/api_key.py

```diff
@@ -0,0 +1,103 @@
+from typing import Optional
+
+from flask import current_app
+import jwt
+from sqlalchemy import Column, ForeignKey, String
+from sqlalchemy.orm import relationship
+
+from pbench.server.database.database import Database
+from pbench.server.database.models import TZDateTime
+from pbench.server.database.models.users import User
+
+
+class APIKeyError(Exception):
+    """A base class for errors reported by the APIKey class."""
+
+    def __init__(self, message):
+        self.message = message
+
+    def __str__(self):
+        return repr(self.message)
+
+
+class APIKeys(Database.Base):
```

Comment:
In all the other DB classes, the class name is *singular*; an instance represents a single "API key", not multiple "API keys". Only the table name is plural!